torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1529,12 +1529,15 @@ void LoopNest::rfactor(
 
   // Store loops below the target point.
   std::vector<const For*> output_loops;
+  bool output_contains_target = false;
 
   while (st) {
     if (For* f = dynamic_cast<For*>(st)) {
       if (f->var() == reduction_var) {
         target_for = f;
-        output_loops.push_back(f);
+      } else if (target_for && !output_contains_target) {
```

Comment:
It would be nice to add some comment or refactor this code a bit (not necessarily in this PR though). It's quite difficult to follow.

Reply:
I think if its difficult to follow now we should prioritize cleaning it up. We're at the point with rfactor that we've bolted enough fixes on it that it's mostly fixes. Let's refactor rfactor.